CMakeLists.txt

```diff
@@ -44,31 +50,33 @@ if(OPENSSL_VERSION VERSION_GREATER_EQUAL 1.1.1)
         add_definitions(-DENABLE_SSL_KEYLOG)
     endif()
 endif()
+
 option(ENABLE_MYSQL "Build with MySQL support" ON)
 if(ENABLE_MYSQL)
     find_package(MySQL REQUIRED)
     include_directories(${MYSQL_INCLUDE_DIR})
     target_link_libraries(trojan ${MYSQL_LIBRARIES})
     add_definitions(-DENABLE_MYSQL)
 endif()
+
 option(FORCE_TCP_FASTOPEN "Force build with TCP Fast Open support" OFF)
 if(FORCE_TCP_FASTOPEN)
     add_definitions(-DTCP_FASTOPEN=23 -DTCP_FASTOPEN_CONNECT=30)
 endif()
+
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL Linux)
```

Comment:
https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_NAME.html

On systems that have the uname command, this variable is set to the output of uname -s, thus not cross-compiling friendly.

Reply:
I prefer to `CMAKE_SYSTEM_NAME`

https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html
https://github.com/openwrt/openwrt/blob/master/include/cmake.mk#L47

Reply:
So should we use [`CMAKE_SYSTEM_NAME`](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html)?

Reply:
Got another method but it doesn't detect Linux directly.

```cmake
if(UNIX AND NOT APPLE)
    # for Linux, BSD, Solaris, Minix
endif()
```

Most of the existing solutions use CMAKE_SYSTEM_NAME

Reply:
I don't like the `UNIX AND NOT APPLE`.

Reply:
Anyway, I force pushed and resolved this.